llvm/lib/Transforms/Scalar/DeadStoreElimination.cpp

```diff
@@ -1162,21 +1172,11 @@ struct DSEState {
                       << *Def->getMemoryInst()
                       << ") is at the end the function \n");
 
-    auto MaybeLoc = getLocForWrite(Def->getMemoryInst());
-    if (!MaybeLoc) {
-      LLVM_DEBUG(dbgs() << "  ... could not get location for write.\n");
-      return false;
-    }
-
+    MemoryLocation DefLoc = *getLocForWrite(Def->getMemoryInst());
```

Comment:
I would feel more comfortable if you passed in this MemoryLocation already computed in the caller instead. That makes it clearer that this really cannot fail.

Reply:
We still want to push the memory access in the worklist though, so we would be still passing the MemoryDef, right? Felt a bit cleaner to pass only one the MemoryDef.

Reply:
Yes, you should pass both the MemoryDef and MemoryLocation.